modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -241,8 +243,9 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
   public altScriptHash?: number;
   public supportAltScriptDestination?: boolean;
   private readonly _network: utxolib.Network;
+  private readonly _amountType: 'number' | 'bigint';
 
-  protected constructor(bitgo: BitGoBase, network: utxolib.Network) {
+  protected constructor(bitgo: BitGoBase, network: utxolib.Network, amountType: 'number' | 'bigint' = 'number') {
```

Comment:
shouldn't we add the type parameter at the class level?

```
export abstract class AbstractUtxoCoin<TNumber extends bigint | number = number> extends BaseCoin`
```

and then for doge we can do `Doge extends AbstractUtxoCoin<bigint>`

Reply:
We could, but this adds quite a few complications because then every single function that takes in `AbstractUtxoCoin` needs to be updated to expect either `AbstractUtxoCoin<bigint>` or `AbstractUtxoCoin<number>`. (I actually tried this first)
Additionally, this isn't necessary because there are no class properties that have to use the `TNumber` generic and we only need to change the interface for a few select functions.

Reply:
ok I see